### ISSUE TITLE
Including <execution> header will break intellisense (with 1.6.0-insiders)

### ISSUE THREAD

**Issue by ziofu** [microsoft/vscode-cpptools#7972]
Issue Type: <b>Bug</b>

Steps to reproduce:
Just include the <execution> header in any file and Intellisense will break.
(Note: c++ standard is forced to c++20 in the configuration settings)

Code:

```
#include <span>
#include <execution>

int main(int, char**) 
{    
    std::span<int> s;  
     s. // typing "s." will not give any suggestion unless #include <execution> is commented
}
```

Extension version: 1.6.0-insiders
VS Code version: Code 1.59.0 (379476f0e13988d90fab105c5c19e7abc8b1dea8, 2021-08-04T23:13:12.822Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz (12 x 3696)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.93GB (25.37GB free)|
|Process Argv|C:\\Users\\ziofu\\Desktop\\test --crash-reporter-id e7bcb6cb-7f5e-4d59-9881-cdc49bf27199|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805cf:30301675
pythonvspyt200:30340761
vscextlangct:30333562
binariesv615:30325510
vsccppwt:30329788
pythonvssor306:30344512
bridge0708:30335490
vstre464cf:30350173
bridge0723:30351747

```

</details>

**Comment by sean-mcmanus** [comment]
I'm not reproing the issue. What version of the C/C++ headers are you using? What IntelliSense mode, i.e. windows-msvc-x64? Can you compare you Log Diagnostics output with the screenshot below?

![image](https://user-images.githubusercontent.com/19859882/129410259-d34f1448-f93f-4d5d-a5fe-a28c5c945e33.png)

Our automated tests also include the execution header for C++20.

**Comment by ziofu** [comment]
Hello,

Indeed the C/C++ log reports:
"IntelliSense process crash detected." but nothing else.

"intelliSenseMode" is "windows-msvc-x64" 

I'm using Visual Studio Version 17.0.0 Preview 3.0
("compilerPath": "C:\\Program Files\\Microsoft Visual Studio\\2022\\Preview\\VC\\Tools\\MSVC\\14.30.30423\\bin\\Hostx64\\x64\\cl.exe",)

I'm attaching the diagnostics log (it seems quite similar to yours)

[log.txt](https://github.com/microsoft/vscode-cpptools/files/6984964/log.txt)

**Comment by Colengms** [comment]
I'm able to repro the issue.  This is the stack I'm getting:

<details>
<summary>Crash stack</summary>

```
>	cpptools-srv.exe!get_locator_for_rescanned_selection_second_operand(a_type * class_struct_union_type, a_rescan_control_block * rcblock, int call_rescan_case, int * followed_by_left_paren, a_symbol_locator * locator, int * err) Line 8658	C++
 	cpptools-srv.exe!scan_field_selection_operator(an_operand * operand_1, a_rescan_control_block * rcblock, int call_rescan_case, int offsetof_case, an_operand * result, an_operand * bound_function_selector) Line 9240	C++
 	cpptools-srv.exe!rescan_expr_with_substitution_internal(an_expr_node * expr, a_rescan_control_block * rcblock, int local_options, an_operand * result, an_operand * bound_function_selector, int top_level_expr) Line 50208	C++
 	cpptools-srv.exe!make_rescan_operand_full(an_expr_node * expr, a_rescan_control_block * rcblock, int local_options, an_operand * operand, an_operand * bound_function_selector) Line 4988	C++
 	cpptools-srv.exe!make_call_rescan_operands(a_rescan_control_block * rcblock, an_operand * operand, an_operand * bound_function_selector, a_source_position * operator_position, unsigned int * operator_tok_seq_number, a_source_position * closing_paren_position) Line 5196	C++
 	cpptools-srv.exe!scan_function_call(an_operand * operand, an_operand * bound_function_selector, a_rescan_control_block * rcblock, an_operand * result) Line 6379	C++
 	cpptools-srv.exe!rescan_expr_with_substitution_internal(an_expr_node * expr, a_rescan_control_block * rcblock, int local_options, an_operand * result, an_operand * bound_function_selector, int top_level_expr) Line 50197	C++
 	cpptools-srv.exe!rescan_expr_with_substitution(an_expr_node * expr, a_type * guide_type, a_rescan_control_block * rcblock, a_constant * constant) Line 50464	C++
 	cpptools-srv.exe!copy_template_param_expr(an_expr_node * expr, a_template_arg * template_arg_list, a_template_param * template_param_list, a_type * guide_type, a_source_position * source_pos, int options, int * copy_error, a_ctws_state * ctws_state, a_constant * constant, a_constant * * alloc_con) Line 18815	C++
 	cpptools-srv.exe!substitute_expr(an_expr_node * expr, const Dyn_array<a_subst_pairs_descr,FE_allocator> & subst_pairs, a_ctws_state * ctws_state, int options, a_constant * cp, a_constant * * p_allocated_cp, int * p_err) Line 25977	C++
 	cpptools-srv.exe!check_requirement_expr(an_expr_node * req_expr, const Dyn_array<a_subst_pairs_descr,FE_allocator> & subst_pairs, a_ctws_state * ctws_state, int constrained, int * p_is_noexcept) Line 26020	C++
 	cpptools-srv.exe!requires_expr_satisfied(an_expr_node * requires_expr, const Dyn_array<a_subst_pairs_descr,FE_allocator> & subst_pairs) Line 26191	C++
 	cpptools-srv.exe!copy_template_param_expr(an_expr_node * expr, a_template_arg * template_arg_list, a_template_param * template_param_list, a_type * guide_type, a_source_position * source_pos, int options, int * copy_error, a_ctws_state * ctws_state, a_constant * constant, a_constant * * alloc_con) Line 19355	C++
 	cpptools-srv.exe!make_rescan_operand_full(an_expr_node * expr, a_rescan_control_block * rcblock, int local_options, an_operand * operand, an_operand * bound_function_selector) Line 4996	C++
 	cpptools-srv.exe!make_rescan_operands(a_rescan_control_block * rcblock, an_operand * operand_1, an_operand * operand_2, an_operand * operand_3, a_source_position * operator_position, unsigned int * operator_tok_seq_number, a_source_position * operator_position_2) Line 5144	C++
 	cpptools-srv.exe!scan_arith_prefix_operator(a_rescan_control_block * rcblock, an_operand * result) Line 12312	C++
 	cpptools-srv.exe!rescan_expr_with_substitution_internal(an_expr_node * expr, a_rescan_control_block * rcblock, int local_options, an_operand * result, an_operand * bound_function_selector, int top_level_expr) Line 50077	C++
 	cpptools-srv.exe!rescan_expr_with_substitution(an_expr_node * expr, a_type * guide_type, a_rescan_control_block * rcblock, a_constant * constant) Line 50464	C++
 	cpptools-srv.exe!copy_template_param_expr(an_expr_node * expr, a_template_arg * template_arg_list, a_template_param * template_param_list, a_type * guide_type, a_source_position * source_pos, int options, int * copy_error, a_ctws_state * ctws_state, a_constant * constant, a_constant * * alloc_con) Line 18815	C++
 	cpptools-srv.exe!constraint_satisfied(an_expr_node * constraint, a_template_arg * template_arg_list, a_template_param * template_param_list, a_diag_list * diag_list, int options, int * p_fatal, int * p_copy_error) Line 25834	C++
 	cpptools-srv.exe!constraint_satisfied(an_expr_node * constraint, a_template_arg * template_arg_list, a_template_param * template_param_list, a_diag_list * diag_list, int options, int * p_fatal, int * p_copy_error) Line 25793	C++
 	cpptools-srv.exe!constraint_satisfied(an_expr_node * constraint, a_template_arg * template_arg_list, a_template_param * template_param_list, a_diag_list * diag_list, int options, int * p_fatal, int * p_copy_error) Line 25793	C++
 	cpptools-srv.exe!requires_constraint_satisfied(a_symbol * template_sym, a_requires_clause * rcp, a_template_arg * args, int diagnose) Line 10652	C++
 	cpptools-srv.exe!check_template_constraints(a_symbol * template_sym, a_template_arg * args, int diagnose) Line 10796	C++
 	cpptools-srv.exe!determine_function_viability(a_symbol * proj_function_symbol, a_symbol * overloaded_function_symbol, int is_template_id, a_template_arg * template_arg_list, a_symbol * surrogate_function_conv_sym, a_type * routine_type, an_init_component * arg_list, int have_selector, an_operand * bound_function_selector, a_type * implicit_selector_type, int ctor_conversion_case, int effects_copy_initialization, int allow_udc_on_arguments, int arg_dep_lookup_done, int from_arg_dep_lookup, int dependent_call, int known_to_be_visible, int is_overloaded_operator, int allow_post_declared_functions, int conv_context, an_overload_context ovl_context, a_candidate_function * * candidate_functions, int * matched_except_for_missing_selector, int * matched_except_for_selector, int * discarded_because_post_decl, int * discarded_restricted_mismatch) Line 6117	C++
 	cpptools-srv.exe!try_overloaded_function_match(a_symbol * overloaded_function_symbol, int is_template_id, a_template_arg * template_arg_list, an_init_component * arg_list, an_init_component * init_list_ctor_arg_list, int have_selector, an_operand * bound_function_selector, int ctor_conversion_case, int effects_copy_initialization, int allow_udc_on_arguments, int arg_dep_lookup_done, int from_arg_dep_lookup, int dependent_call, int ignore_templates, int known_to_be_visible, int is_overloaded_operator, int conv_context, an_overload_context ovl_context, a_candidate_function * * candidate_functions, a_symbol * * inaccessible_match, int * matched_except_for_missing_selector, int * matched_except_for_selector, int * discarded_restricted_mismatch) Line 6690	C++
 	cpptools-srv.exe!conversion_to_class_possible(an_operand * source_operand, an_init_component * alep, a_type * dest_type, int try_bitwise_copy, int is_copy_initialization, int orig_is_copy_initialization, a_type * ref_binding_type, int is_direct_binding, int conv_context, a_conv_descr * conversion, a_conv_descr * ctor_arg_conversion, int * ambiguous, a_candidate_function * * ambiguity_list) Line 19783	C++
 	cpptools-srv.exe!user_defined_conversion_possible(an_operand * source_operand, a_type * dest_type, int need_lvalue_result, int is_copy_initialization, int orig_is_copy_initialization, a_type * ref_binding_type, int is_direct_binding, int conv_context, a_conv_descr * conversion, a_conv_descr * ctor_arg_conversion, int * failed) Line 20743	C++
 	cpptools-srv.exe!conversion_possible(an_operand * source_operand, a_type * dest_type, int * is_transparent, a_type * orig_dest_type, int need_lvalue_result, int is_copy_initialization, int orig_is_copy_initialization, a_type * ref_binding_type, int is_direct_binding, int conv_context, an_error_code incompatible_err, a_source_position * err_pos, a_conv_descr * conversion, a_conv_descr * ctor_arg_conversion) Line 21021	C++
 	cpptools-srv.exe!prep_elision_initializer_operand(an_operand * source_operand, a_type * dest_type, int fill_in_dtor, int conv_context, an_error_code err_code, int * elision_done, a_dynamic_init * * dip) Line 22926	C++
 	cpptools-srv.exe!scan_return_expression(a_type * required_type, an_error_code err_code, a_dynamic_init * * dip, an_init_component * * alep) Line 47699	C++
 	cpptools-srv.exe!return_statement() Line 6940	C++
 	cpptools-srv.exe!statement(int is_dependent_statement, int marked_as_gnu_extension) Line 7706	C++
 	cpptools-srv.exe!compound_statement_full(int at_function_level, int explicit_return_type, int is_catch_clause, int is_statement_expr, int marked_as_gnu_extension, a_type * * p_result_type) Line 8154	C++
 	cpptools-srv.exe!statement(int is_dependent_statement, int marked_as_gnu_extension) Line 7647	C++
 	cpptools-srv.exe!dependent_statement() Line 3630	C++
 	cpptools-srv.exe!switch_statement() Line 4364	C++
 	cpptools-srv.exe!statement(int is_dependent_statement, int marked_as_gnu_extension) Line 7660	C++
 	cpptools-srv.exe!compound_statement_full(int at_function_level, int explicit_return_type, int is_catch_clause, int is_statement_expr, int marked_as_gnu_extension, a_type * * p_result_type) Line 8154	C++
 	cpptools-srv.exe!scan_function_body(a_routine * rout_ptr, a_func_info_block * func_info, unsigned long flags, a_macro_arg_fixup * * macro_arg_fixup_cache, a_macro_arg_fixup * * end_of_macro_arg_fixup_cache) Line 2057	C++
 	cpptools-srv.exe!function_prototype_instantiation(a_symbol * template_sym) Line 6232	C++
 	cpptools-srv.exe!template_declaration(a_tmpl_decl_state * decl_state) Line 31442	C++
 	cpptools-srv.exe!template_or_specialization_declaration_full(a_tmpl_decl_state * decl_state, int is_generic, a_decl_parse_state * orig_dps) Line 34523	C++
 	cpptools-srv.exe!template_or_specialization_declaration(a_token_kind * final_token, int export_present, a_source_position * export_pos, int is_generic, a_decl_parse_state * orig_dps) Line 34319	C++
 	cpptools-srv.exe!template_directive_or_declaration(a_token_kind * final_token, int options, a_source_position * directive_start_pos) Line 40633	C++
 	cpptools-srv.exe!check_special_declaration_form(a_decl_parse_state * state, a_token_kind * final_token) Line 20641	C++
 	cpptools-srv.exe!scan_nonmember_declaration(a_decl_parse_state * dps, a_source_range * linkage_spec_range_ptr) Line 21711	C++
 	cpptools-srv.exe!declaration(int function_definition_allowed, int is_old_style_param_decl, int is_top_level_declaration, int marked_as_gnu_extension, a_param_id * param_id_list, a_source_range * linkage_spec_range_ptr) Line 22073	C++
 	cpptools-srv.exe!namespace_declaration(a_token_kind * final_token, int in_nested_namespace_decl, int is_inline, a_source_position * nested_namespace_pos, a_symbol * * ns_definition_sym) Line 16149	C++
 	cpptools-srv.exe!check_special_declaration_form(a_decl_parse_state * state, a_token_kind * final_token) Line 20661	C++
 	cpptools-srv.exe!scan_nonmember_declaration(a_decl_parse_state * dps, a_source_range * linkage_spec_range_ptr) Line 21711	C++
 	cpptools-srv.exe!declaration(int function_definition_allowed, int is_old_style_param_decl, int is_top_level_declaration, int marked_as_gnu_extension, a_param_id * param_id_list, a_source_range * linkage_spec_range_ptr) Line 22073	C++
 	cpptools-srv.exe!translation_unit() Line 22148	C++
 	cpptools-srv.exe!process_translation_unit(const char * file_name, int is_primary, an_exported_template_file * exported_file) Line 785	C++
 	cpptools-srv.exe!cfe_main(int argc, char * * argv) Line 128	C++
 	cpptools-srv.exe!cfe_main_exception_handler(int argc, char * * argv) Line 604	C++
 	cpptools-srv.exe!edg_main(int argc, char * * argv) Line 220	C++
 	cpptools-srv.exe!invoke_edge_compiler(std::vector<std::string,std::allocator<std::string>> & arguments) Line 332	C++
 	cpptools-srv.exe!edge_compiler_main(int argc, const char * * argv) Line 401	C++
 	cpptools-srv.exe!preparse(int argc, const char * * argv, a_scout_store * store, edge::translation_unit * trans_unit) Line 2719	C++
 	cpptools-srv.exe!a_compiler_thread::compiler_thread_routine(a_compiler_thread * ct) Line 15210	C++
 	[External Code]	
 	cpptools-srv.exe!thread_start<unsigned int (__stdcall*)(void *),1>(void * const parameter) Line 97	C++
 	[External Code]	
 	kernel32.dll![Frames below may be incorrect and/or missing, no symbols loaded for kernel32.dll]	Unknown
```
</details>

**Comment by sean-mcmanus** [comment]
Yeah, the repro requires the latest 14.30.30423 headers. Our tests are based on the older headers. Looks like it might be worth adding a new test with the latest headers, i.e. there could be other crashes or new squiggle errors caused by other headers.

**Comment by Colengms** [comment]
It also repro's in VS.  I opened a bug internally against VS.

**Comment by sean-mcmanus** [comment]
Adding a simple check for null fixes the crash, so we should be able to release a fix for 1.6.0-insiders2 (whether or not the VS side can provide a "better" fix).

**Comment by Colengms** [comment]
This should be fixed in 1.6.0-insiders2.
